ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -424,17 +424,70 @@ bracketChainSyncClient
     acquireContext lopBucket cschState (CSJEnabledConfig jumpSize) = do
         tid <- myThreadId
         atomicallyWithMonotonicTime $ \time -> do
-          initialGsmState <- getGsmState
-          updateLopBucketConfig lopBucket initialGsmState time
+          gsmState <- getGsmState
+          initializeLopBucketConfig lopBucket gsmState time
           cschJumpInfo <- newTVar Nothing
           context <- Jumping.makeContext varHandles jumpSize tracerCsj
-          Jumping.registerClient context peer cschState $ \cschJumping -> ChainSyncClientHandle
-            { cschGDDKill = throwTo tid DensityTooLow
-            , cschOnGsmStateChanged = updateLopBucketConfig lopBucket
-            , cschState
-            , cschJumping
-            , cschJumpInfo
-            }
+          Jumping.registerClient gsmState context peer cschState $ \cschJumping ->
+            ChainSyncClientHandle
+              { cschGDDKill = throwTo tid DensityTooLow
+              , cschOnGsmStateChanged = updateLopBucketConfig lopBucket
+                  -- The GSM only transitions out of CaughtUp if the tip of its
+                  -- selection is much older than expected (eg 20 minutes).
+                  -- There are many possible explanations for why that could
+                  -- have happened, so it's not obvious what is the best thing
+                  -- to do for 'CaughtUpPreSyncing'.
```

Comment:
One other potential scenario: Someone ran a (caught-up) Daedalus wallet and closed their laptop lib, suspending their machine. Then, when they wake up again after a ~week. In that case, using CSJ would be beneficial.

However, I am not sure if this is a likely scenario; maybe Daedalus already takes care of restarting the node in this case.